### PR TITLE
Separated scripts from modules into /scripts

### PR DESCRIPTION
**Pull request by wu-ming233**
Separated `main(args)` from the following modules/files:
- `src/rank_llm/retrieve/pyserini_retriever.py`
- `src/rank_llm/analysis/estimate_costs.py`
- `src/rank_llm/analysis/response_analysis_verbose.py`
- `src/rank_llm/evaluation/trec_eval.py`

Each into a separate file prefixed with `run_{file_name}` under `src/rank_llm/scripts`.

Additionally, some lines under `demo` and `rerank` were reformatted by black, along with some updates in imports for the separated modules above.

### CODE REVIEW

**Comment by sahel-sh** [comment]
@wu-ming233 thank you for the PR, please resolve the conflicting files.

**Comment by wu-ming233** [comment]
> @wu-ming233 thank you for the PR, please resolve the conflicting files.

Updated!

File: src/rank_llm/analysis/response_analysis_verbose.py [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
 import json
 import os
 import re
-from typing import List, Dict
-
 import sys
-import os
+from typing import Dict, List
 
 SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
```
Comment (sahel-sh) [inline src/rank_llm/analysis/response_analysis_verbose.py:7]: This block of  adding path is no longer needed, after separating scripts.

File: src/rank_llm/evaluation/trec_eval.py [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
 import tempfile
 
+import pandas as pd
 from pyserini.search import get_qrels_file
 from pyserini.util import download_evaluation_script
 
-import sys
-import os
-
 SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
```
Comment (sahel-sh) [inline src/rank_llm/evaluation/trec_eval.py:11]: similarly, no longer needed

File: src/rank_llm/retrieve/pyserini_retriever.py [hunk 3]
```diff
@@ -1,28 +1,27 @@
...
-    get_topics,
     get_qrels,
+    get_topics,
 )
 from tqdm import tqdm
 
-import sys
-import os
-
 SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
```
Comment (sahel-sh) [inline src/rank_llm/retrieve/pyserini_retriever.py:25]: similarly no longer needed

File: src/rank_llm/scripts/run_trec_eval.py
Comment (sahel-sh) [inline src/rank_llm/scripts/run_trec_eval.py:1]: - This file is missing adding args and calling main().
- The conflicting PR adds a new arg for the rerank_results_dir that should be also integrated here.

**Review by sahel-sh — CHANGES_REQUESTED** [review]

**Comment by wu-ming233** [comment]
Removed unnecessary sys path snippets, and incorporated the new PR an hour ago.

**Review by sahel-sh — APPROVED** [review]

**Pull request merged**